From 26142079ef861c7cc4b88f5093e11d5bc10c0f17 Mon Sep 17 00:00:00 2001
From: Kevin Peng <kevinp@marvell.com>
Date: Tue, 15 May 2018 16:36:24 +0800
Subject: [PATCH 0644/1051] dts: armada-ap806: reserve 16M memory for tee

Reserve 16M memory for TEE, starts from 0x4400000.

These memory are used for TEE code, data, communication channel
and shared memory with Non-Trusted world.

For the TEE code and data section, it will be configured unacceable
to Non-Trusted World by memory controller.

The rest memories are either used for communication or shared memory
between Trusted-World and Non-trusted World, which is TEE implementation
defined and may be managed by the TEE driver.

Change-Id: I32d02516ca1d1226ba724d50c5994fed7c06c3d4
Signed-off-by: Kevin Peng <kevinp@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06.
Just some minor context mods in order to port to wrlinux]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 7b05bee74ff8..06b9cca405ee 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -47,6 +47,17 @@
 		};
 	};
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		tee@4400000 {
+			reg = <0 0x4400000 0 0x1000000>;
+			no-map;
+		};
+	};
+
 	ap806 {
 		#address-cells = <2>;
 		#size-cells = <2>;
-- 
2.17.1

